### ISSUE TITLE
draw by timeout

### ISSUE THREAD

**Issue by gahtan-syarif** [Disservin/fast-chess#410]
![Screenshot 2024-05-18 040204](https://github.com/Disservin/fast-chess/assets/155860115/74cb934b-12b2-4178-87a9-48cb79577e5d)
apparently this is an actual rule, cutechess seems to already have this while fast-chess not yet (i think). probably would rarely if ever come up but might be worth implementing nonetheless

**Comment by gahtan-syarif** [comment]
i guess this is more of a chess-library pr but ig can be posted here too :p

**Comment by Disservin** [comment]
How is this a chess lib thing ? Chess lib doesn’t know anything about time control and there is a public function for insufficient material check I think

**Comment by gahtan-syarif** [comment]
i think the function exists within chess lib's board.hpp, in fastchess it extracts the data from the board object

**Comment by gahtan-syarif** [comment]
yep https://github.com/Disservin/chess-library/blob/c45383444969339f5df1457dc4089a3a89d702a3/src/board.hpp#L639 the one in chess lib can only check if both sides have insufficient material, in this case one side is insufficient while the opponent can still mate

**Comment by gahtan-syarif** [comment]
apparently, theres also a draw by disconnect in cutechess
